Race Condition inside PulsarClient  at Datanode

<!-- Please state your issue using the following template and, most importantly, in English. -->

#### Steps/Code to reproduce:

    --- FAIL: TestDataNodeTtChannel/test_expire_allocation_after_receiving_tt_msg (0.20s)
        testing.go:1042: race detected during execution of test
    testing.go:1042: race detected during execution of test

#### Expected result:
     The test should be passed without any failures.

#### Actual results:
Detailed Race information is shown here:
   WARNING: DATA RACE
Read at 0x00c003795ca8 by goroutine 908:
  github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).internalClose()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:854 +0x687
  github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).runEventsLoop()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:808 +0x29b

Previous write at 0x00c003795ca8 by goroutine 909:
  github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).grabConn()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:970 +0x17c6
  github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).reconnectToBroker()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:883 +0x238
  github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).runEventsLoop.func2()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:787 +0xab

Goroutine 908 (running) created at:
  github.com/apache/pulsar-client-go/pulsar.newPartitionConsumer()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:208 +0xf3a
  github.com/apache/pulsar-client-go/pulsar.(*consumer).internalTopicSubscribeToPartitions.func1()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_impl.go:326 +0x778

Goroutine 909 (running) created at:
  github.com/apache/pulsar-client-go/pulsar.(*partitionConsumer).runEventsLoop()
      /go/pkg/mod/github.com/apache/pulsar-client-go@v0.5.0/pulsar/consumer_partition.go:780 +0x146


#### Environment:
- Milvus version(e.g. v2.0.0-RC2 or 8b23a93):
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:

--- a/go.mod
+++ b/go.mod
@@ -50,7 +50,7 @@ require (
 )
 
 replace (
-	github.com/apache/pulsar-client-go => github.com/apache/pulsar-client-go v0.5.0
+	github.com/apache/pulsar-client-go => github.com/apache/pulsar-client-go v0.6.0
 	google.golang.org/grpc => google.golang.org/grpc v1.38.0
 	github.com/keybase/go-keychain => github.com/99designs/go-keychain v0.0.0-20191008050251-8e49817e8af4
 )

--- a/internal/util/mqclient/id.go
+++ b/internal/util/mqclient/id.go
@@ -14,4 +14,16 @@ package mqclient
 type MessageID interface {
 	// Serialize the message id into a sequence of bytes that can be stored somewhere else
 	Serialize() []byte
+
+	// Get the message ledgerID
+	LedgerID() int64
+
+	// Get the message entryID
+	EntryID() int64
+
+	// Get the message batchIdx
+	BatchIdx() int32
+
+	// Get the message partitionIdx
+	PartitionIdx() int32
 }

--- a/internal/util/mqclient/pulsar_id.go
+++ b/internal/util/mqclient/pulsar_id.go
@@ -21,8 +21,28 @@ type pulsarID struct {
 	messageID pulsar.MessageID
 }
 
+// Check if pulsarID implements pulsar.MessageID and MessageID interface
+var _ pulsar.MessageID = &pulsarID{}
+var _ MessageID = &pulsarID{}
+
 func (pid *pulsarID) Serialize() []byte {
 	return pid.messageID.Serialize()
+}
+
+func (pid *pulsarID) LedgerID() int64 {
+	return pid.messageID.LedgerID()
+}
+
+func (pid *pulsarID) EntryID() int64 {
+	return pid.messageID.EntryID()
+}
+
+func (pid *pulsarID) BatchIdx() int32 {
+	return pid.messageID.BatchIdx()
+}
+
+func (pid *pulsarID) PartitionIdx() int32 {
+	return pid.messageID.PartitionIdx()
 }
 
 func SerializePulsarMsgID(messageID pulsar.MessageID) []byte {

--- a/internal/util/mqclient/pulsar_id_test.go
+++ b/internal/util/mqclient/pulsar_id_test.go
@@ -27,6 +27,11 @@ func TestPulsarID_Serialize(t *testing.T) {
 	binary := pid.Serialize()
 	assert.NotNil(t, binary)
 	assert.NotZero(t, len(binary))
+
+	pid.LedgerID()
+	pid.EntryID()
+	pid.BatchIdx()
+	pid.PartitionIdx()
 }
 
 func Test_SerializePulsarMsgID(t *testing.T) {

--- a/internal/util/mqclient/rmq_id.go
+++ b/internal/util/mqclient/rmq_id.go
@@ -21,8 +21,31 @@ type rmqID struct {
 	messageID rocksmq.UniqueID
 }
 
+// Check if rmqID implements MessageID interface
+var _ MessageID = &rmqID{}
+
 func (rid *rmqID) Serialize() []byte {
 	return SerializeRmqID(rid.messageID)
+}
+
+func (rid *rmqID) LedgerID() int64 {
+	// TODO
+	return 0
+}
+
+func (rid *rmqID) EntryID() int64 {
+	// TODO
+	return 0
+}
+
+func (rid *rmqID) BatchIdx() int32 {
+	// TODO
+	return 0
+}
+
+func (rid *rmqID) PartitionIdx() int32 {
+	// TODO
+	return 0
 }
 
 func SerializeRmqID(messageID int64) []byte {

--- a/internal/util/mqclient/rmq_id_test.go
+++ b/internal/util/mqclient/rmq_id_test.go
@@ -25,6 +25,11 @@ func TestRmqID_Serialize(t *testing.T) {
 	bin := rid.Serialize()
 	assert.NotNil(t, bin)
 	assert.NotZero(t, len(bin))
+
+	rid.LedgerID()
+	rid.EntryID()
+	rid.BatchIdx()
+	rid.PartitionIdx()
 }
 
 func Test_SerializeRmqID(t *testing.T) {

